__tests__/components/TransactionList/TransactionRow.spec.ts

```diff
@@ -0,0 +1,888 @@
+                        new Mosaic(new MosaicId(networkMock.currency.mosaicIdHex), UInt64.fromUint(100)),
+                    ]),
+                },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            jest.spyOn(vm, 'isIncomingTransaction').mockReturnValue(true);
+
+            // Act:
+            const amount = vm.getAmount();
+            const amountColor = vm.getAmountColor();
+            const hasNetworkMosaic = vm.hasNetworkMosaic();
+            const getAmountMosaicId = vm.getAmountMosaicId();
+            const isAmountShowTicker = vm.isAmountShowTicker();
+
+            // Assert:
+            expect(amount).toBe(100);
+            expect(amountColor).toBe('green');
+            expect(hasNetworkMosaic).toBe(true);
+            expect(getAmountMosaicId.toHex()).toBe('091F837E059AE13C');
+            expect(isAmountShowTicker).toBe(false);
+        });
+
+        test('display message icon', () => {
+            // Arrange:
+            const wrapper = getTransactionRowWrapper(
+                { currentProfile: testnetProfile },
+                {
+                    transaction: mockTransferTransaction(
+                        [new Mosaic(new MosaicId('461A9811A1DC1FBB'), UInt64.fromUint(0))],
+                        PlainMessage.create('test-message'),
+                    ),
+                },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // Act:
+            const hasMessage = vm.hasMessage();
+
+            // Assert:
+            expect(hasMessage).toBe(true);
+            expect(vm.messagePayload).toBe('test-message');
+            expect(wrapper.find('.extend-icon-holder.message').exists()).toBe(true);
+        });
+
+        test('hide message icon', () => {
+            // Arrange:
+            const wrapper = getTransactionRowWrapper({ currentProfile: testnetProfile }, { transaction: mockTransferTransaction() });
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // Act:
+            const hasMessage = vm.hasMessage();
+
+            // Assert:
+            expect(hasMessage).toBe(false);
+            expect(wrapper.find('.extend-icon-holder.message').exists()).toBe(false);
+        });
+
+        test('returns encrypted message title when message type is EncryptedMessage', () => {
+            // Arrange:
+            const recipient = getTestAccount('remoteTestnetRecipient');
+
+            const wrapper = getTransactionRowWrapper(
+                { currentProfile: testnetProfile },
+                {
+                    transaction: mockTransferTransaction(
+                        [],
+                        EncryptedMessage.create('test', recipient.publicAccount, currentSigner.privateKey),
+                    ),
+                },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // Act + Assert:
+            expect(vm.messagePayload).toBe('Encrypted message');
```

Comment:
should there be icon(s) for encypted message ?

Reply:
y. there is still an icon, which I will add it into assert.